### PR TITLE
Adds sounds for the PDA

### PR DESCRIPTION
**Pull request by Azarak**
## About The Pull Request
This is a port from Citadel. ~~I've changed the pen noises to be played locally for the user so pulling a traitorous pen doesn't give you away by sound.~~ https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8181

This makes the PDA play all sorts of nice sounds, aswell as sounds for ejecting or inserting the ID/pen/cartridge.
The noises for pressing buttons will need for your ringer to be ON, the other noises get played either way.

## Why It's Good For The Game
Audio feedback is nice, plus those sounds really fit!

## Changelog
:cl:
add: PDA now has sounds for interacting with it, aswell as inserting or ejecting items, such as the ID
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -445,6 +445,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 //BASIC FUNCTIONS===================================
 
 			if("Refresh")//Refresh, goes to the end of the proc.
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:449]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 2]
```diff
@@ -459,12 +461,16 @@ GLOBAL_LIST_EMPTY(PDAs)
 						font_mode = FONT_ORBITRON
 					if (MODE_VT)
 						font_mode = FONT_VT
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:465]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 3]
```diff
@@ -459,12 +461,16 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if ("Change_Color")
 				var/new_color = input("Please enter a color name or hex value (Default is \'#808000\').",background_color)as color
 				background_color = new_color
 
 			if ("Toggle_Underline")
 				underline_flag = !underline_flag
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:473]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 4]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
 					mode = round(mode/10)
 					if(mode==4 || mode == 5)//Fix for cartridges. Redirects to hub.
 						mode = 0
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:483]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 5]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if ("Authenticate")//Checks for ID
 				id_check(U)
 			if("UpdateInfo")
 				ownjob = id.assignment
 				if(istype(id, /obj/item/card/id/syndicate))
 					owner = id.registered_name
 				update_label()
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_processing.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:492]:
```suggestion
					playsound(src, 'sound/machines/terminal_processing.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 6]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
+					addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, src, 'sound/machines/terminal_success.ogg', 15, 1), 13)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:493]:
```suggestion
					addtimer(CALLBACK(GLOBAL_PROC, .proc/playsound, src, 'sound/machines/terminal_success.ogg', 15, TRUE), 1.3 SECONDS)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 7]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if("Eject")//Ejects the cart, only done from hub.
 				eject_cart(U)
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_eject.ogg', 50, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:497]:
```suggestion
					playsound(src, 'sound/machines/terminal_eject.ogg', 50, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 8]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 
 //MENU FUNCTIONS===================================
 
 			if("0")//Hub
 				mode = 0
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:504]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 9]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if("1")//Notes
 				mode = 1
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:508]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 10]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if("3")//Atmos scan
 				mode = 3
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
 			if("4")//Redirects to hub
 				mode = 0
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:524]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
			if("21")//Read messeges
				mode = 21
				if(!silent)
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
			if("3")//Atmos scan
				mode = 3
				if(!silent)
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
			if("4")//Redirects to hub
				mode = 0
				if(!silent)
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 11]
```diff
@@ -473,41 +479,64 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 				toggle_light(U)
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
 			if("Medical Scan")
 				if(scanmode == PDA_SCANNER_MEDICAL)
 					scanmode = PDA_SCANNER_NONE
 				else if((!isnull(cartridge)) && (cartridge.access & CART_MEDICAL))
 					scanmode = PDA_SCANNER_MEDICAL
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:539]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
			if("Medical Scan")
				if(scanmode == PDA_SCANNER_MEDICAL)
					scanmode = PDA_SCANNER_NONE
				else if((!isnull(cartridge)) && (cartridge.access & CART_MEDICAL))
					scanmode = PDA_SCANNER_MEDICAL
				if(!silent)
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 12]
```diff
@@ -518,6 +547,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 					scanmode = PDA_SCANNER_NONE
 				else if((!isnull(cartridge)) && (cartridge.access & CART_ENGINE))
 					scanmode = PDA_SCANNER_HALOGEN
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:551]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 13]
```diff
@@ -531,13 +562,17 @@ GLOBAL_LIST_EMPTY(PDAs)
 					scanmode = PDA_SCANNER_NONE
 				else if((!isnull(cartridge)) && (cartridge.access & CART_ATMOS))
 					scanmode = PDA_SCANNER_GAS
+				if(!silent)
+					playsound(src, 'sound/machines/terminal_select.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:566]:
```suggestion
					playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 14]
```diff
@@ -531,13 +562,17 @@ GLOBAL_LIST_EMPTY(PDAs)
...
 			if("Drone Phone")
 				var/alert_s = input(U,"Alert severity level","Ping Drones",null) as null|anything in list("Low","Medium","High","Critical")
 				var/area/A = get_area(U)
 				if(A && alert_s && !QDELETED(U))
 					var/msg = "<span class='boldnotice'>NON-DRONE PING: [U.name]: [alert_s] priority alert in [A.name]!</span>"
 					_alert_drones(msg, TRUE, U)
 					to_chat(U, msg)
+					if(!silent)
+						playsound(src, 'sound/machines/terminal_success.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:575]:
```suggestion
						playsound(src, 'sound/machines/terminal_success.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 15]
```diff
@@ -663,6 +698,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 	id = null
 	updateSelfDialog()
 	update_icon()
+	playsound(src, 'sound/machines/terminal_eject.ogg', 50, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:701]:
```suggestion
	playsound(src, 'sound/machines/terminal_eject.ogg', 50, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 16]
```diff
@@ -716,6 +752,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 	// If it didn't reach, note that fact
 	if (!signal.data["done"])
 		to_chat(user, "<span class='notice'>ERROR: Server isn't responding.</span>")
+		if(!silent)
+			playsound(src, 'sound/machines/terminal_error.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:756]:
```suggestion
			playsound(src, 'sound/machines/terminal_error.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 17]
```diff
@@ -733,6 +771,8 @@ GLOBAL_LIST_EMPTY(PDAs)
 	// Log in the talk log
 	user.log_talk(message, LOG_PDA, tag="PDA: [initial(name)] to [target_text]")
 	to_chat(user, "<span class='info'>PDA message sent to [target_text]: \"[message]\"</span>")
+	if(!silent)
+		playsound(src, 'sound/machines/terminal_success.ogg', 15, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:775]:
```suggestion
		playsound(src, 'sound/machines/terminal_success.ogg', 15, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 18]
```diff
@@ -855,6 +895,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 		to_chat(user, "<span class='notice'>You remove [inserted_item] from [src].</span>")
 		inserted_item = null
 		update_icon()
+		user.playsound_local(src, 'sound/machines/pda_button2.ogg', 50, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:898]:
```suggestion
		user.playsound_local(src, 'sound/machines/pda_button2.ogg', 50, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 19]
```diff
@@ -886,6 +927,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 			return FALSE
 		insert_id(I, user)
 		update_icon()
+		playsound(src, 'sound/machines/pda_button1.ogg', 50, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:930]:
```suggestion
		playsound(src, 'sound/machines/pda_button1.ogg', 50, TRUE)
```

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 20]
```diff
@@ -973,6 +1020,7 @@ GLOBAL_LIST_EMPTY(PDAs)
 			to_chat(user, "<span class='notice'>You slide \the [C] into \the [src].</span>")
 			inserted_item = C
 			update_icon()
+			user.playsound_local(user, 'sound/machines/pda_button1.ogg', 50, 1)
```
Comment (Rohesie) [inline code/game/objects/items/devices/PDA/PDA.dm:1023]:
```suggestion
			user.playsound_local(user, 'sound/machines/pda_button1.ogg', 50, TRUE)
```

**Review by Rohesie — DISMISSED** [review]

**Comment by Acer202** [comment]
Honestly it would be funny to have people so paranoid that they freak out when someone ejects a pen. Too much pranking potential. I would urge you to make it audible to others!

**Comment by Azarak** [comment]
> 
> 
> Honestly it would be funny to have people so paranoid that they freak out when someone ejects a pen. Too much pranking potential. I would urge you to make it audible to others!

Agreed! Will do it

**Pull request merged**